packages/odyssey-storybook/src/components/odyssey-mui/Button/Button.stories.tsx

```diff
@@ -12,54 +12,162 @@
 
 import type { Meta, StoryObj } from "@storybook/react";
 
-import { Button, AddIcon } from "@okta/odyssey-react-mui";
+import { Button, Icon, iconDictionary } from "@okta/odyssey-react-mui";
 import type { ButtonProps } from "@okta/odyssey-react-mui";
 import { MuiThemeDecorator } from "../../../../.storybook/components/MuiThemeDecorator";
 
 import { userEvent, within } from "@storybook/testing-library";
 import { expect } from "@storybook/jest";
 
+const iconOptions = ["None", ...Object.keys(iconDictionary)];
+const iconMapping = Object.fromEntries(
+  iconOptions.map((value) => [
+    value,
+    value !== "None" ? (
+      <Icon
+        name={value as keyof typeof iconDictionary}
+        label={value}
+        size="medium"
+      />
+    ) : null,
+  ])
+);
+
+type ButtonStoryProps = Omit<ButtonProps, "startIcon"> & {
```

Comment:
We need to modify the type just for the stories, to enable the icons to be `ReactElement<Icon>` in the component but `string` in the controls.